src/clusterfuzz/_internal/swarming/__init__.py

```diff
@@ -0,0 +1,135 @@
+  try:
+    _get_new_task_spec(command, job_name, '')
+    return True
+  except ValueError:
+    return False
+
+
+def _get_task_name():
+  return 't-' + str(uuid.uuid4()).lower()
+
+
+def _get_swarming_config():
+  """Returns the swarming config."""
+  return local_config.SwarmingConfig()
+
+
+def _get_new_task_spec(command: str, job_name: str,
+                       download_url: str) -> swarming_pb2.NewTaskRequest:
+  """Gets the configured specifications for a swarming task."""
+  job = data_types.Job.query(data_types.Job.name == job_name).get()
+  config_name = job.platform
+  swarming_config = _get_swarming_config()
+  instance_spec = swarming_config.get('mapping').get(config_name, None)
+  if instance_spec is None:
+    raise ValueError(f'No mapping for {config_name}')
+  swarming_pool = swarming_config.get('swarming_pool')
+  swarming_realm = swarming_config.get('swarming_realm')
+  # The priority of the task
+  priority = instance_spec['priority']
+  # The command to launch the startup script
+  startup_command = instance_spec['command']
+  # The cas instance storing the startup script
+  cas_instance = instance_spec['cas_instance']
+  # The startup script archive hash
+  digest_hash = instance_spec['digest_hash']
+  # The startup script size in bytes
+  digest_size_bytes = instance_spec['digest_size_bytes']
+  # The service account that the task runs as.
+  service_account = instance_spec['service_account_email']
+  # If this task request slice is not scheduled after waiting this long,
+  # the task state will be set to EXPIRED.
+  expiration_secs = instance_spec['expiration_secs']
+  # Maximum number of seconds the task can run before its process is
+  # forcibly terminated and the task results in TIMED_OUT.
+  execution_timeout_secs = instance_spec['execution_timeout_secs']
+  if command == 'fuzz':
+    execution_timeout_secs = swarming_config.get('fuzz_task_duration')
+  new_task_request = swarming_pb2.NewTaskRequest(
+      name=_get_task_name(),
+      priority=priority,
+      realm=swarming_realm,
+      service_account=service_account,
+      task_slices=[
+          swarming_pb2.TaskSlice(
+              expiration_secs=expiration_secs,
+              properties=swarming_pb2.TaskProperties(
+                  command=[startup_command],
+                  dimensions=[
+                      swarming_pb2.StringPair(key='os', value=job.platform),
+                      swarming_pb2.StringPair(key='pool', value=swarming_pool)
+                  ],
+                  cas_input_root=swarming_pb2.CASReference(
+                      cas_instance=cas_instance,
+                      digest=swarming_pb2.Digest(
+                          hash=digest_hash, size_bytes=digest_size_bytes)),
+                  execution_timeout_secs=execution_timeout_secs,
+                  env=[
+                      swarming_pb2.StringPair(key='UWORKER', value='True'),
+                      swarming_pb2.StringPair(key='SWARMING_BOT', value='True'),
```

Comment:
What code will be starting the task on the batch side?

Reply:
You mean on the swarming side right?
This is to be figured out once we receive the machines and start setting things up. It will be similar to the existing startup scripts for linux/mac